polyglot.js

```diff
@@ -2,27 +2,56 @@ class PolyGlot {
+				}
+		}
+
+		static get languages() {
+			switch(game.system.id.toLowerCase()){
+				default:
+					return CONFIG[game.system.id.toUpperCase()].languages;
+				case "wfrp4e":
+					let re = /Language \((.+)\)/i;
+					const pack = game.packs.get("wfrp4e.skills");
+					return pack.getIndex().then( itemList => {
+						let langs = {};
+						for (let item of itemList){
+							let found = item.name.match(re);
+							if(found && found.length > 1 && found[1].length >= 1){
+								langs[found[1]] = found[0];
+							}
+						}
+						Promise.resolve(langs);
+					});
+			}
+		}
 
 	renderChatLog(chatlog, html, data) {
-		let lang_html = $(`
-		<div class="polyglot-lang-select flexrow">
-			<label>Language : </label>
-			<select name="polyglot-language">
-			</select>
-		</div>
-		`);
-		html.find("#chat-controls").after(lang_html)
+		let lang_html = ``;
+		switch ( game.system.id.toLowerCase() ){
+			default:
+				lang_html = $(`
+				<div id="polyglot"  class="polyglot-lang-select flexrow">
+					<label>Language : </label>
+					<select name="polyglot-language">
+					</select>
+				</div>
+				`);
+				break;
+			case "wfrp4e":
+				lang_html = $(`
+				<div id="polyglot"  class="polyglot-lang-select-wfrp4e flexrow">
```

Comment:
This whole thing would be much nicer with this as the only change : 
```js
const select_class = game.system.id.toLowerCase() === "wfrp4e" ? "polyglot-lang-select-wfrp4e" : "polyglot-lang-select";
const lang_html = $(`
		<div class="${select_class} flexrow">
....
`);
```
(and yes, I used `let` for the lang_html variable, it should have been `const` from the start.)

However, I am not at all a fan here of having the class of the DOM element be different just for the language. I understand that you're doing this for the CSS to work only for wfrp4e, but that seems wrong. The select should not require any special styling at all. Isn't WFRP4E already applying its own custom styling to all the `select` elements ? If it doesn't, then its' an issue in the system because it means any other select is going to be affected.
Can you explain why exactly polyglot needs to apply any kind of special handling/css of the select for wfrp4e?

Reply:
no WFRP4E is not applying its own custom styling to all `select` elements.  When I first added polyglot it was not styled at all.

Reply:
Ok. I think in that case, I'd prefer to not have any custom CSS. Let the select appear ugly in WFRP until the system itself sets the proper CSS for all selects. That's the safer way forward.